### DISCUSSION TITLE
Run snippets only when in certain language code blocks?

### DISCUSSION THREAD

**Discussion by MitchellHarrison** [artisticat1/obsidian-latex-suite#319] [Q&A] [answered]
Hello all! I'm building a workflow for my algorithms courses in Obsidian, and I already use LaTeX suite for my math stuff. There's a great plug-in for writing LaTeX algorithms in psudocode blocks like the following:

```
```pseudo
\begin{algorithm} 
\caption{My first algo} 
\begin{algorithmic}
\State $X = 10$
\State Letting $Y = 10$, we can see things
\If{X = 10}
	\State Do a thing for now
	\Comment{This will do the thing we discussed}
\EndIf
\end{algorithmic}
\end{algorithm}
```

I know there is a setting for LaTeX suite to make specific languages be read as math blocks called `Code languages to interpret as math mode`, but I'm wondering if anyone knows how I might check specifically that I am in a `pseudo` block without calling it a math block? I have snippets that I want to work in `pseudo` blocks that interfere with my normal math ones. Any help would be amazing! 

Thanks in advance.

**Comment by superle3** [comment]
Am thinking about implementing that but not sure how api/syntax would work and it would add another option to the many options.
Like only a string or an array or even Regex.

**Comment by superle3** [comment, marked as answer]
you can now with https://github.com/artisticat1/obsidian-latex-suite/pull/503 as of 1.11.0
with for example in python

```ts
const snippet = {trigger: "npp", replacement: "import numpy as np", options: "c", language: "python"}
```
